packages/agreement-process/src/utilities/errorMappers.ts

```diff
@@ -120,7 +120,7 @@ export const suspendAgreementErrorMapper = (
   error: ApiError<ErrorCodes>
 ): number =>
   match(error.code)
-    .with("agreementNotFound", () => HTTP_STATUS_NOT_FOUND)
+    .with("agreementNotFound", "documentNotFound", () => HTTP_STATUS_NOT_FOUND)
```

Comment:
In this case `documentNotFound` is an internal server error
```suggestion
    .with("agreementNotFound", () => HTTP_STATUS_NOT_FOUND)
```